Add options for sharing resources UI

anushkavidanage opened the pull request:
# Pull Request Details

## What issue does this PR address

- Add options to select Access mode and Recipient type in UI

- Link to associated issue: https://github.com/anusii/solidpod/issues/288

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [x] Pre-exisiting lint errors noted: [HERE]
- [x] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome on Android
  - [ ] Chrome
  - [ ] iOS
  - [x] Linux
  - [ ] MacOS
  - [x] Windows
- [ ] Added 2 reviewers (or 1 for private repositories then they add another)

## Finalising

Once PR discussion is complete and 2 reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev

anushkavidanage committed:
add optional parameters

cdawei reviewed (approved):
The changes look good. Thanks @anushkavidanage.

gjwgit commented:
Thanks @anushkavidanage. Please merge and I'll bump the version and release.

It could do with a documentation update too?

anushkavidanage committed:
Documentation, changelog, and version update

anushkavidanage commented:
> Thanks @anushkavidanage. Please merge and I'll bump the version and release.
> 
> It could do with a documentation update too?

Updated Readme and changed changelog along with pubspec. Will merge now.
